src/js/playlist/source.js

```diff
@@ -31,6 +32,13 @@ define([
         if (! _source.type) {
             var extension = strings.extension(_source.file);
             _source.type = extensionmap.extType(extension);
+
+            if (utils.isYouTube(_source.file, _source.type)) {
+                _source.type = 'youtube';
+            }
+            if (utils.isRtmp(_source.file, _source.type)) {
+                _source.type = 'rtmp';
+            }
         }
 
         if (!_source.type) {
```

Comment:
The normalize type cases:
1. ```if (_source.type && _source.type.indexOf('/') > 0) {``` only applies to the user setting the type as a mime type. (file extensions won't have slashes, so this could probably go before line 32.

2. m3u8 and smil are file extensions or the user might set these. Guess it's fine at the end.